src/plot_api/plot_api.js

```diff
@@ -2157,6 +2148,37 @@ Plotly.update = function update(gd, traceUpdate, layoutUpdate, traces) {
     });
 };
 
+// empty out types for all axes containing these traces
+// so we auto-set them again
+var axLetters = ['x', 'y', 'z'];
+function clearAxisTypes(gd, traces, layoutUpdate) {
+    if(typeof traces === 'number') traces = [traces];
+    else if(!Array.isArray(traces) || !traces.length) {
```

Comment:
Do we really need this?

`traces` should be already sanatize [`helpers.coerceTraceIndices`](https://github.com/plotly/plotly.js/blob/790d470b901a38713728b5c9dc10bc624fed39ad/src/plot_api/helpers.js#L448) called [here](https://github.com/plotly/plotly.js/blob/790d470b901a38713728b5c9dc10bc624fed39ad/src/plot_api/plot_api.js#L1323).

Reply:
I suppose I can move the `helpers.coerceTraceIndices` call into `restyle` and `update` instead of `_restyle`, then we can get rid of this.